### PR TITLE
Refactor tests to remove reliance on applicationBuilder

### PR DESCRIPTION
**Pull request by james-work-account**

### CODE REVIEW

File: app/controllers/sections/items/BaseItemsNavigationController.scala [hunk 1]
```diff
@@ -57,4 +61,22 @@ trait BaseItemsNavigationController extends BaseNavigationController {
       case None =>
         Future.successful(Redirect(routes.ItemsIndexController.onPageLoad(request.ern, request.draftId)))
     }
+
+  def withCnCodeInformation(idx: Index)(f: CnCodeInformation => Result)(implicit request: DataRequest[_]): Future[Result] = {
```
Comment (LewisMitchell) [inline app/controllers/sections/items/BaseItemsNavigationController.scala:65]: Could you add a test for this please?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Reviewed app package

File: .g8/checkboxPage/generated-test/controllers/$className$ControllerSpec.scala [hunk 2]
```diff
@@ -1,138 +1,91 @@
...
       MockUserAnswersService.set().returns(Future.successful(emptyUserAnswers))
 
-      val application =
-        applicationBuilder(userAnswers = Some(emptyUserAnswers))
-          .overrides(
-            bind[Navigator].toInstance(new FakeNavigator(onwardRoute)),
-            bind[UserAnswersService].toInstance(mockUserAnswersService)
-          )
-          .build()
+      val result = controller.onSubmit(testErn, testDraftId, NormalMode)(request.withFormUrlEncodedBody(("value", "answer")))
```
Comment (LewisMitchell) [inline .g8/checkboxPage/generated-test/controllers/$className$ControllerSpec.scala:62]: Was the form body wrong before?

File: .g8/checkboxPage/generated-test/controllers/$className$ControllerSpec.scala [hunk 3]
```diff
@@ -1,138 +1,91 @@
...
-      val application = applicationBuilder(userAnswers = Some(userAnswers)).build()
-      
-      running(application) {
-        val request = FakeRequest(GET, $className;format="decap"$Route)
-
-        val view = application.injector.instanceOf[$className$View]
-
-        val result = route(application, request).value
+    "must populate the view correctly on a GET when the question has previously been answered" in new Test(Some(
+      emptyUserAnswers.set($className$Page, "answer")
```
Comment (LewisMitchell) [inline .g8/checkboxPage/generated-test/controllers/$className$ControllerSpec.scala:50]: Was the page answer wrong before?

File: .g8/checkboxPage/generated-test/controllers/$className$ControllerSpec.scala (hunk 2 above)
Comment (james-work-account) [reply .g8/checkboxPage/generated-test/controllers/$className$ControllerSpec.scala:62]: Nope, copy/paste error

File: .g8/checkboxPage/generated-test/controllers/$className$ControllerSpec.scala (hunk 3 above)
Comment (james-work-account) [reply .g8/checkboxPage/generated-test/controllers/$className$ControllerSpec.scala:50]: Nope, copy/paste error

File: .g8/checkboxPage/generated-test/controllers/$className$ControllerSpec.scala [hunk 4]
```diff
@@ -1,138 +1,91 @@
...
+    "must populate the view correctly on a GET when the question has previously been answered" in new Test(Some(
+      emptyUserAnswers.set($className$Page, $className$.values.toSet)
+    )) {
+      val result = controller.onPageLoad(testErn, testDraftId, NormalMode)(request)
 
-        status(result) mustEqual OK
-        contentAsString(result) mustEqual view(form.fill($className$.values.toSet), NormalMode)(dataRequest(request), messages(application)).toString
-      }
+      status(result) mustEqual OK
+      contentAsString(result) mustEqual view(form.fill("answer"), NormalMode)(dataRequest(request, userAnswers.get), messages).toString
```
Comment (LewisMitchell) [inline .g8/checkboxPage/generated-test/controllers/$className$ControllerSpec.scala:55]: This can be changed back in a later PR

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**